### PR TITLE
feat: rewrite WebDAV in Go for optimized memory usage

### PR DESCRIPTION
**Pull request by bouassaba**

### CODE REVIEW

**Comment by dsonck92** [comment]
Nice progress already, probably you want to convert this to draft for the time being to mark it a little more explicit.

You definitely went the route I also had in mind when seeing the WebDav server was written in TypeScript/JavaScript. I still have hopes that it can be memory efficient even on that engine, but in practice I haven't seen it yet. So a change to Go seems logical.

**Comment by bouassaba** [comment]
@dsonck92 Thanks! I will convert it to draft ;)

File: webdav/.env [hunk 1]
```diff
@@ -1,3 +1,15 @@
 PORT=8082
+
+# Security
+SECURITY_API_KEY="7znl9Zd8F!4lRZA43lEQb757mCy"
```
Comment (loboda4450) [inline webdav/.env:4]: Take it as a note:

I'd rethink the idea of passing the api keys, jwt etc. as a envs. It is a good practice to roll them from time to time, but having it that way will make everybody forget about it. I will create a issue for that.

**Review by loboda4450 — APPROVED** [review]

File: webdav/.env (hunk 1 above)
Comment (bouassaba) [reply webdav/.env:4]: I agree, thanks!

**Comment by dsonck92** [comment]
@bouassaba do you want to add the linters before or after this PR? You can probably copy it over from the existing files, and enable as many linters as possible to see how they like the code.

**Comment by bouassaba** [comment]
> @bouassaba do you want to add the linters before or after this PR? You can probably copy it over from the existing files, and enable as many linters as possible to see how they like the code.

@dsonck92  sure I will copy them afterwards.

**Comment by dsonck92** [comment]
Alright, then I will have a look at the changes if there's anything strange going on, and ignore the formatting for now.

**Pull request merged**